api/v1alpha1/httproute_types.go

```diff
@@ -263,12 +266,27 @@ type ForwardToTarget struct {
 	// Support: Core
 	//
 	// +optional
-	TargetPort *TargetPort `json:"targetPort" protobuf:"bytes,2,opt,name=targetPort"`
+	TargetPort *TargetPort `json:"targetPort,omitempty" protobuf:"bytes,2,opt,name=targetPort"`
+
+	// Weight specifies the proportion of traffic to be forwarded to a targetRef.
+	// A valid weight value is 0-100. The sum of weights across targetRefs must
+	// equal 100. If only one targetRef is specified, weight is ignored and all
+	// traffic is forwarded to targetRef. Weight is divided equally across all
+	// targetRefs that do not specify a weight.
+	//
+	// Support: Core
+	//
+	// +optional
+	Weight *TargetWeight `json:"weight,omitempty" protobuf:"bytes,3,opt,name=weight"`
```

Comment:
Each selected target could have their respective set of changes to request and response headers that need to take effect.

Reply:
@yiyangy understood. I am only including the `weight` attribute to keep the PR small and focused. Adding post routing filters, i.e. header manipulation, can be added in a follow-on PR.